### PR TITLE
SNOW-783493: Implement show

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/mock/mock_plan.py [hunk 1]
```diff
@@ -76,7 +77,9 @@ def __init__(
         self.source_plan = source_plan
         self.child = child
         self.expr_to_alias = {}
-        self.queries = []
+        mock_query = MagicMock()
+        mock_query.sql = "select 1"
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/mock/mock_plan.py:81]: Let's use a flag value, something more unique like `SELECT MOCK_TEST_FAKE_QUERY()`

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-stan — APPROVED** [review]

File: src/snowflake/snowpark/mock/mock_plan.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/snowpark/mock/mock_plan.py:81]: good idea!

File: tests/integ/scala/test_dataframe_suite.py [hunk 2]
```diff
@@ -910,6 +913,7 @@ def test_sample_on_union(session):
     )
 
 
+@pytest.mark.localtest
```
Comment (sfc-gh-stan) [inline tests/integ/scala/test_dataframe_suite.py:916]: I know this is from the existing code but why are we only calling `DataFrame.show()` without asserting anything. Let's improve this by asserting captured stdout like https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function

Comment (sfc-gh-aling) [reply tests/integ/scala/test_dataframe_suite.py:916]: updated

**Pull request merged**